cassandane/Cassandane/Cyrus/JMAPMailbox.pm

```diff
@@ -5282,4 +5296,64 @@ sub test_mailbox_ignore_notes_subfolders
     $self->assert_str_equals($inboxId, $res->[1][1]{list}[0]{id});
 }
 
+sub test_mailbox_set_create_specialuse_nochildren
+    :min_version_3_7 :needs_component_jmap :NoStartInstances
+{
+    my ($self) = @_;
+
+    $self->{instance}->{config}->set('specialuse_nochildren' => '\\Trash');
+    $self->_start_instances();
+    $self->_setup_http_service_objects();
+    $self->setup_default_using();
+
+    # set up a Trash folder with \Trash special-use annotation
+    my $imaptalk = $self->{store}->get_client();
+    $imaptalk->create("Trash", "(USE (\\Trash))");
```

Comment:
I think the JMAP test should use JMAP as much as possible, rather than using IMAP. So this should be

    [Mailbox/set, {
      create => {
        trashmbox => {
           name => 'Trash',
           role => 'trash'
        }
      }
    }, 'R1']

and then you can read the trash mailbox id from the response, rather than parsing the IMAP response as a dlist.

Reply:
Oh great, thanks.  I wasn't sure how to do these operations over JMAP, and I already had IMAP equivalents in the other test, so it was easy enough to just do the same again.  But I'll have a go at changing them to use pure JMAP.